File: src/test/java/no/utgdev/diy/ex3/Service.java
package no.utgdev.diy.ex3;

import no.utgdev.diy.annotations.Inject;

import java.util.List;

import static java.util.Arrays.asList;

public class Service {
    @Inject
    private HelloDao helloDao;

    @Inject
    private WorldDao worldDao;

    public List<String> fragments() {
        return asList(
                helloDao.hello(),
                worldDao.world()
        );
    }
}

File: src/test/kotlin/no/utgdev/diy/ex3/Service.kt
package no.utgdev.diy.ex3

import java.util.*
import no.utgdev.diy.annotations.Inject;

class Service {
    @Inject
    private val helloDao: HelloDao? = null

    @Inject
    private val worldDao: WorldDao? = null
    fun fragments(): List<String?> {
        return Arrays.asList(
            helloDao!!.hello(),
            worldDao!!.world()
        )
    }
}